Ignore value of Release header when comparing spec files

OlegGirko opened the pull request:
Replacing release taken from source RPM metadata with `@RELEASE@` placeholder in `Release:` header of spec file before comparing doesn't work if there is a variable substitution in this header.

For example, you may have this in your project config:
```
Release: <CI_CNT>%%{?dist}.<B_CNT>
```
This will cause `Release:` header look like this in generated spec file:
```
Release: 1%{?dist}.2
```
whereas release extracted from source RPM metadata will look like `1.fc27.2`, and these substitutions will not work:
```sh
sed -i -e "s,Release:.*${ver_rel_old}$,Release: @RELEASE@," old/$file
sed -i -e "s,Release:.*${ver_rel_new}$,Release: @RELEASE@," new/$file
```
This change makes comparing srpm files to ignore value of `Release:` header. This value is generated from project config anyway.

olafhering commented:
declare -A requires a new bash, SLE11 will fail.

OlegGirko commented:
Actually, I have a better idea than this workaround. Why do we need to check for version in `Release:` header at all?
* We know that old and new source RPM packages we compare were built in OBS.
* We know that spec file inside these source RPM packages is the spec file that was used to build these packages.
* As a result, we know that spec file can not contain anything wrong in `Release:` header.

Hence, instead of trying to substitute variables in `Release:` header, why not just replace
```
sed -i -e "s,Release:.*${ver_rel_old}$,Release: @RELEASE@," old/$file
sed -i -e "s,Release:.*${ver_rel_new}$,Release: @RELEASE@," new/$file
```
with
```
sed -i -e 's,^Release:.*$,Release: @RELEASE@,' old/$file
sed -i -e 's,^Release:.*$,Release: @RELEASE@,' new/$file
```
?

olafhering commented:
I think dealing with Release: handles the case where the `Release:` string really changes, even if the pkg content remains identical. In this case the pkg should be considered changed.

OlegGirko commented:
Yes, but `Release:` header is autogenerated anyway, so it changes every time package is rebuilt. Even if the original intent was to check whether `Release:` header changed in the spec file in checked-in sources, current implementation does not implement it: `Release:` header's value in the spec file in *generated* source RPM is always the same as release field in package's metadata, unless there is variable substitution in *generated* `Release:` header.

olafhering commented:
I wonder if we can just throw away Release: in the *.spec files. Any changes in the pattern would be caught in binary package.

OlegGirko commented:
I think, we don't need to check value of `Release:` header at all. It's completely irrelevant in *original* spec file because it's replaced with autogenerated content in *generated* spec file. This content is generated from `Release:` header in project config, so the only reason for change in release generation pattern is change in project config.

olafhering commented:
Yes, please update the pull request accordingly.

OlegGirko commented:
Sorry, I was distracted by other stuff and forgot about this PR for some time. Now I'm returning back to it.

I've updated my change according to discussion above, rebased it on current `master` branch and changed this PR's description to reflect that. Now it's ready to be merged.
